src/sentry/static/sentry/app/views/ruleEditor/utils.js

```diff
@@ -0,0 +1,33 @@
+/*eslint no-use-before-define:0*/
+
+export function getHtmlForNode(node) {
+  let {label, formFields} = node;
+  let pattern = /{(\w+)}/g;
+  return label.replace(pattern, (str, match) => {
+    const fieldData = formFields[match];
+    if (fieldData) {
+      return getField(match, fieldData);
+    } else {
+      return str;
+    }
+  });
+}
+
+function getField(name, data) {
+  const getFieldTypes = {
+    choice: getChoiceField,
+    number: getInputField.bind(null, 'number'),
+    string: getInputField.bind(null, 'text'),
+  };
+  return getFieldTypes[data.type](name, data);
+}
+
+function getChoiceField(name, data) {
+  return `<select name="${name}">
```

Comment:
ooof, Why can't it just return a react component? i.e, can't this just be jsx?

Reply:
Definitely. Will tackle the UI changes in another PR though. For now, I'd leave the UI implementation as-is and just create here the exact string that was previously being returned by the API.